docs/naming_conventions.md

```diff
@@ -0,0 +1,71 @@
+
+# Naming Conventions in Kids First FHIR Data Model
+
+This document describes the naming conventions for ids and
+the file names for resources in the Kids First FHIR data model.
+
+## Resource Id
+
+1. A resource in the FHIR model should always have the `id` attribute
+populated by model developers. We should never let the server auto populate
+`id` since it is the primary way that machines and humans will remember,
+reference, and lookup a resource.
+
+2. The `id` value should conform to the the regex pattern in the
+[FHIR spec](https://www.hl7.org/fhir/datatypes.html#id).
+
+3. All resource `id` values should have the same casing. There does not seem
+   to be a consistent standard or recommendation for this but we should pick
+   one and stick to it. Based on observations, the options are:
+   - Camel case
+   - Snake case with hyphens instead of underscores
+
+TODO - Choose casing
+
+## Resource File Name
+
+1. All resource types EXCEPT `StructureDefinition` should follow this
+ file naming pattern:
+
+```
+# Pattern
+<resource type><delimiter><resource id>.json
+
+# Example
+Patient_participant.json
+```
+
+2. All `StructureDefinition` resources should follow this file naming pattern:
+
+```
+# Pattern
+StructureDefinition_<base type>_<resource id>.json
+
+# Example
+StructureDefinition_Patient_participant.json
+StructureDefinition_Extension_age-at-event.json
+```
+
+- Easy to identify extensions vs other conformance resources.
+- Easy to visually scan and sort for resources by type
+
+### File Name Delimiter
+The delimiter should NOT be any of the characters in the valid character set
+for `id`. Could be one of the following [ `_`, ...?]
```

Comment:
I was thinking underscore would be a good delimiter? Anyone have other thoughts?

Reply:
As long as they don't get used in the resource type.
Though, since you mention it, an ellipsis would be kinda neat.